app/classes/site_data.rb

```diff
@@ -401,45 +354,52 @@ def load_field_counts(field, user_id = nil)
       @user_data[usr_id.to_i][field] = cnt.to_i
     end
   end
+  # rubocop:enable Metrics/MethodLength
 
   # Load all the stats for a given User.  (Load for all User's if none given.)
   #
   #   load_user_data(user.id)
   #   user.contribution = @user_data[user.id][:metric]
   #
   def load_user_data(id = nil)
-    if id
-      @user_id = id.to_i
-      users = [User.find(id)]
-    else
-      @user_id = nil
-      users = User.all
-    end
+    return unless id
+
+    # if id
+    #   @user_id = id.to_i
+    #   user = [User.find(id)]
+    # else
+    #   @user_id = nil
+    #   users = User.all
+    # end
+
+    @user_id = id.to_i
+    user = User.find(id)
 
     # Prime @user_data structure.
     @user_data = {}
```

Comment:
This is fishy.  Either this is never called on multiple users, in which case @user_data doesn't need to be a hash, or it *is* called on multiple users, in which case shouldn't this is `@user_data ||= {}`?  As it is, it clobbers the hash each time and sticks a single user in it.